lib/python/Screens/FlashManager.py

```diff
@@ -0,0 +1,600 @@
+										unlink("%s/%s" % (rootFolder, fileName))
+						except:
+							print("[FlashManager] postFlashActionCallback: failed to create restore mode flagfile")
+				self.startDownload()
+			else:
+				self.abort()
+		else:
+			self.abort()
+
+	def saveEPG(self):
+		from enigma import eEPGCache
+		epgcache = eEPGCache.getInstance()
+		epgcache.save()
+
+	def startDownload(self, reply=True):
+		self.show()
+		if reply:
+			if "://" in self.source:
+				from Tools.Downloader import downloadWithProgress
+				self["header"].setText(_("Downloading Image"))
+				self["info"].setText(self.imagename)
+				self["summary_header"].setText(self["header"].getText())
+				self.downloader = downloadWithProgress(self.source, self.zippedimage)
+				self.downloader.addProgress(self.downloadProgress)
+				self.downloader.addEnd(self.downloadEnd)
+				self.downloader.addError(self.downloadError)
+				self.downloader.start()
+			else:
+				self.unzip()
+		else:
+			self.abort()
+
+	def downloadProgress(self, current, total):
+		self["progress"].setValue(int(100 * current / total))
+
+	def downloadError(self, reason, status):
+		self.downloader.stop()
+		self.session.openWithCallback(self.abort, MessageBox, _("Error during downloading image\n%s\n%s") % (self.imagename, reason), type=MessageBox.TYPE_ERROR, simple=True)
+
+	def downloadEnd(self):
+		self.downloader.stop()
+		self.unzip()
+
+	def unzip(self):
+		self["header"].setText(_("Unzipping Image"))
+		self["summary_header"].setText(self["header"].getText())
+		self["info"].setText("%s\n%s" % (self.imagename, _("Please wait")))
+		self["progress"].hide()
+		self.delay.callback.remove(self.confirmation)
+		self.delay.callback.append(self.doUnzip)
+		self.delay.start(0, True)
+
+	def doUnzip(self):
+		try:
+			ZipFile(self.zippedimage, "r").extractall(self.unzippedimage)
+			self.doFlashImage()
+		except:
+			self.session.openWithCallback(self.abort, MessageBox, _("Error during unzipping image\n%s") % self.imagename, type=MessageBox.TYPE_ERROR, simple=True)
+
+	def doFlashImage(self):
+		self["header"].setText(_("Flashing Image"))
+		self["summary_header"].setText(self["header"].getText())
+
+		def findimagefiles(path):
+			for path, subdirs, files in walk(path):
+				if not subdirs and files:
+					return checkImageFiles(files) and path
+		imagefiles = findimagefiles(self.unzippedimage)
+		if imagefiles:
+			self.ROOTFSSUBDIR = "none"
+			bootSlots = MultiBoot.getBootSlots()
+			if bootSlots:
+				self.MTDKERNEL = bootSlots[self.multibootslot]["kernel"].split("/")[2]
+				if bootSlots[self.multibootslot].get("ubi"):
+					self.MTDROOTFS = bootSlots[self.multibootslot]["device"]
+				else:
+					self.MTDROOTFS = bootSlots[self.multibootslot]["device"].split("/")[2]
+				if MultiBoot.hasRootSubdir():
+					self.ROOTFSSUBDIR = bootSlots[self.multibootslot]["rootsubdir"]
+			else:
+				self.MTDKERNEL = BoxInfo.getItem("mtdkernel")
+				self.MTDROOTFS = BoxInfo.getItem("mtdrootfs")
+			CMD = [OFGWRITE, OFGWRITE]
```

Comment:
CMD is not a global variable so normal camelCase should be used.  In this case the name might be "cmd".